sklearn/utils/__init__.py

```diff
@@ -819,6 +819,33 @@ def tosequence(x):
         return list(x)
 
 
+def to_object_array(sequence):
```

Comment:
We should keep the utility private to scikit-learn and call it `_to_object_array` with leading underscore.